### PR TITLE
mypy self check: Use mypy version from requirements file

### PR DESCRIPTION
**Pull request by srittau**

### CODE REVIEW

File: tests/mypy_self_check.py [hunk 1]
```diff
@@ -2,17 +2,29 @@
...
+
+
+def determine_mypy_version() -> str:
+    with open(REQUIREMENTS_FILE) as f:
+        for line in f:
+            for regexp in MYPY_REQUIREMENTS_REGEXPS:
+                m = re.match(regexp, line)
+                if m:
+                    return line.strip()
+    raise RuntimeError(f"no mypy version found in {REQUIREMENTS_FILE}")
```
Comment (Akuli) [inline tests/mypy_self_check.py:22]: Can we just install the whole `requirements-tests-py3.txt`? It still wouldn't be the slowest thing in our CI.

Comment (srittau) [reply tests/mypy_self_check.py:22]: Fine by me and it would make this less error prone.

Comment (srittau) [reply tests/mypy_self_check.py:22]: @Akuli I just wanted to make that change, but noticed that after the installation of mypy, mypy's test-requirements.txt file is installed, which may contain conflicting package versions. I think it's safer to just install mypy.

Comment (Akuli) [reply tests/mypy_self_check.py:22]: The code still seems unnecessarily complicated to me. Currently `.github/workflows/tests.yml` contains commands like `pip install $(grep black requirements-tests-py3.txt)`. Maybe we could similarly switch to a simple `if "mypy" in line`?

That said, this is just a small nit, and I'm fine with it either way.

Comment (srittau) [reply tests/mypy_self_check.py:22]: I was considering using something simple, but a simple check would match comments and things like "flake8-mypy". (The latter is less of a concern at the moment, though.)

Comment (Akuli) [reply tests/mypy_self_check.py:22]: Actually, `"mypy" in line` breaks because the file now has a comment mentioning mypy.

**Review by Akuli — APPROVED** [review]

**Comment by srittau** [comment]
@Akuli Actually I just tested that we don't need the mypy requirements file, since we are just executing mypy. So we should be able to use our own requirements file without problems.

**Pull request merged**